crates/bevy_asset/src/lib.rs

```diff
@@ -1902,7 +1902,7 @@ mod tests {
         dir.insert_asset_text(Path::new(a_path), a_ron);
 
         let mut app = App::new();
-        let (gated_memory_reader, gate_opener) = GatedReader::new(MemoryAssetReader { root: dir });
+        let (gated_memory_reader, _gate_opener) = GatedReader::new(MemoryAssetReader { root: dir });
```

Comment:
This is still not quite right. Just remove the `GatedReader::new` entirely, and I think it should work.